Motortest

bonzer35 opened the pull request.

bonzer35 committed:
Added motor testing code - ramps up and then ramps down

bonzer35 committed:
Added motor testing code - ramps up and then ramps down

bonzer35 committed:
Changed max spin speed to 550 (plus 50)

bonzer35 commented:
In progress. Resolution required for UI callback after motor test finished

bonzer35 committed:
Fixed issue with UI not updating 'Motor Test' status once test has finished

Commit:
Merge Main, Resolve Conflicts
- Update 5160 library to 1.1 release, which adds the new velocity reached methods.

bonzer35 committed:
Re-generated KiwiBoard.emf

futurexdesign commented on KiwiBoard.emf:
```diff
@@ -375,7 +375,7 @@
       "parentId": 15,
       "type": "analogItem",
-      "defaultValue": "4",
+      "defaultValue": "false",
```
This was likely a merge error, default value should probably go back to 4

bonzer35 replied:
I don't recall adjusting this

futurexdesign commented on src/KiwiBoardFirmware_main.cpp:
```diff
@@ -247,6 +247,10 @@ void CALLBACK_FUNCTION dry(int id) {
     run(2, &menuDry);
 }
 
+void CALLBACK_FUNCTION test(int id) {
```
Lets rename this `motorTest` or similar so that it is less ambiguous

bonzer35 replied:
Yep - good idea

futurexdesign commented on src/motorControl.cpp:
```diff
@@ -5,6 +5,8 @@
 #include "motorControl.h"
 #include "picoPlatform.h"
 
+bool motorTest = 0;
```
If this is needed, it should be moved into the `state` structure.

futurexdesign commented on src/motorControl.cpp:
```diff
@@ -199,9 +224,17 @@ void MotorControl::exec() {
+        Serial.print("Current Speed: ");
+        Serial.println(motor->getCurrentSpeed());
+        if(motor->isTargetVelocityReached())
+            stopMotion();
```
Perhaps instead of calling `stopMotion`, you could bypass that call, and instead set the velocity to zero in this code.   That may allow you to bypass some of the extra changes made in the `exec` check.   When you check if the velocity hits zero, reset the state variables.  Essentially bypass the various `isStopping` checks.

bonzer35 replied:
Good idea - thanks

futurexdesign commented on src/motorControl.cpp:
```diff
@@ -199,9 +224,17 @@ void MotorControl::exec() {
 
     // If past end time...
-    if (millis() >= state.run_end) {
+    if (!motorTest && millis() >= state.run_end) {
```
Is this check for motorTest necessary?

bonzer35 replied:
Without it, the motor gets stopped before it's started

futurexdesign commented on src/splash_BACKUP.h:
Can you please remove this file

bonzer35 replied:
Sure

Commit:
Fix merge issue where 2 menu options got assigned the same ID.

Commit:
Merge remote-tracking branch 'origin/motortest' into motortest

# Conflicts:
#	src/KiwiBoardFirmware_menu.cpp

Commit:
Fix merge issue where 2 menu options got assigned the same ID.

bonzer35 committed:
Cleaned up code as requested
Renamed motor test menu items to motorTest
Removed call to stopMotion() for the motor status routine

futurexdesign commented:
At a quick glance it seems okay.  I’ll try to look at the code in the ide and built/test tomorrow.

Commit:
Minor Code Cleanup.
